fili-core/src/main/java/com/yahoo/bard/webservice/table/availability/ImmutableAvailability.java

```diff
@@ -76,6 +76,26 @@ public ImmutableAvailability(
     }
 
     /**
+     * Constructor.
+     *
+     * @param tableName  The name of the data source associated with this ImmutableAvailability
+     * @param dimensionIntervals  The dimension availability map by dimension name
+     * @param metricIntervals  The metric availability map
+     * @param dimensionDictionary  The dictionary to resolve dimension names against
+     */
+    public ImmutableAvailability(
+            TableName tableName,
+            Map<String, Set<Interval>> dimensionIntervals,
+            Map<String, Set<Interval>> metricIntervals,
+            DimensionDictionary dimensionDictionary
+    ) {
+        this(
+                tableName,
+                buildAvailabilityMap(dimensionIntervals, metricIntervals, dimensionDictionary)
+        );
+    }
+
```

Comment:
We are replacing `ImmutableAvailability` with `ConcreteAvailability` so do we still wan to fix it here?